api/pkg/apis/v1alpha1/providers/stage/script/script.go

```diff
@@ -112,17 +112,29 @@ func downloadFile(scriptFolder string, script string, stagingFolder string) erro
 	}
 	tPath := filepath.Join(stagingFolder, script)
 
-	out, err := os.Create(tPath)
+	resp, err := http.Get(sPath)
 	if err != nil {
 		return err
 	}
-	defer out.Close()
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
```

Comment:
I think we can add retry based on the status ok

Reply:
Do you mean retrying until get StatusOk?

Reply:
I propose creating another enhancement issue to make retries a common setting on StageSpec (like Schedule) and handle retries in a central place (so that all stages can be retried if needed).

Reply:
An issue to track the enhancement: https://github.com/eclipse-symphony/symphony/issues/381